### ISSUE TITLE
Local Instance should have more types of data (stars, reading log counts)

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#10153]
### Proposal

The desired outcome is that either the default developer instance data dump be extended to include a set of data (reading log entries, community review tags, star ratings, lists) or that a script be created that randomly generates some of this seed data, such that the default admin user on the dev instance (on the initial docker `up`)

### Justification

Allows more community members to contribute

Today, when one launches a local developer instance, to the best of my knowledge, the database doesn't include key data like:
* reading log entries
* community reviews/tags
* star ratings
* lists
that may make it easy for contributors to work on Open Library issues.

<hr>

#### Instructions for Contributors

Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by phoenix7815** [comment]
Hey @mekarpeles , Can I take up this issue?

**Comment by jimchamp** [comment]
Removing https://github.com/internetarchive/openlibrary/labels/Good%20First%20Issue until this issue is broken down.

**Comment by jimchamp** [comment]
As I understand things, we do not want this new data to be included when the tables are initially populated.  This will make things a bit more complicated in regards to the list data, but not too bad for somebody that's comfortable with `psql`.

Here are the general requirements:
- [ ] Create a new `patron_data.sql` file in the `/openlibrary/scripts/dev-instance` directory.  This file will include the [DML](https://en.wikipedia.org/wiki/Data_manipulation_language) which inserts the dummy data.
- [ ] Include instructions for running the script.  This can be as simple as a header comment in `patron_data.sql`.
- [ ] All additional requirements are met (see checkboxes below).

### Reading Log Data
#### Requirements
- [ ] Add at least one book to the "Want to Read" shelf
- [ ] Add at least one book to the "Currently Reading" shelf
- [ ] Add at least one book to the "Already Read" shelf

#### Guidance
- Reading log entries can be found in the `BOOKSHELVES_BOOKS` table
- The `pg_dump` utility can be used to generate the DML for this table
- [Here](https://openlibrary.org/help/faq/reading-log#reading-log) are instructions for adding books to reading log shelves

### Star Ratings
#### Requirements
- [ ] Add star ratings to at least two books

#### Guidance
- Star ratings can be found in the `RATINGS` table
- `pg_dump` can be used to generate the DML for this table
- [Here](https://openlibrary.org/help/faq/reviews#star-ratings) are instructions for star rating a book


### Community Review Tags
#### Requirements
- [ ] Add one or two review tags each to two or three books

#### Guidance
- Review data can be found in the `OBSERVATIONS` table
- `pg_dump` can be used to generate the DML for this table
- [Here](https://openlibrary.org/help/faq/reviews#book-tags) are instructions for adding review tags to books

### Lists
#### Requirements
- [ ] Create one new list with an appropriate name.  Adding a description is optional.
- [ ] Ensure that the list contains at least one work and at least one edition.
- [ ] Ensure that the list contains 5 entries
- [ ] DML for list data should not set explicit values for `ID` or `THING_ID` (more information below)

#### Guidance
- [Here](https://openlibrary.org/help/faq/reading-log#lists) are instructions for creating and managing lists on Open Library.
- Lists containing at least 5 entries are eligible for display on the `/lists` page
- List data are stored in two tables:
  - The `THING` table stores the list's key and other metadata about the list in a single entry
  - The `DATA` table stores the list's contents, name, and description
    - There will be multiple entries for your list in this table --- one for each revision
      - DML for each revision should be included in the `patron_data.sql` script
      - `DATA.THING_ID` is a foreign key that references the `ID` in the `THING` table
- `pg_dump` __cannot__ exclusively be used for this
  - The `THING` table sets `ID` based on a sequence
  - DML produced by `pg_dump` will include an explicit value for `ID`, which may result in a primary key violation if the developer has added other entries to the `THING` table
  - You'll probably want to first insert into the `THING` table explicitly setting the `key`, `type`, and `latest_revision` values.
  - Then, create the `DATA` DML by setting `THING_ID` with a subquery or variable.

I've added the https://github.com/internetarchive/openlibrary/labels/Good%20First%20Issue label to this again, and will assign the next person who asks to contribute.

**Comment by schu96** [comment]
@jimchamp I'd like to be assigned to this issue

**Comment by jimchamp** [comment]
I don't think that I mentioned this, but don't include any works or editions that are not initially included with the dev instance of OL to your DML.

**Comment by schu96** [comment]
Hey @jimchamp, I'm just about finished with this issue but have run into a few ruts:

1. Lists created from running ```patron_data.sql``` does not appear within the user's created lists until the memcache container is restarted or another action updates the memcache (i.e. adding another entry to the list). 
Should I  include a step in the instructions to run ```docker restart openlibrary-memcached-1``` if users don't see the list on the My Books page, or is there a better way to handle this?

2. I currently have a shell script that closely follows the ```create-dev-db-pgdump.sh``` instructions but found that I am also able to use this command to populate the database: 
```docker compose exec -T db psql --quiet -U openlibrary openlibrary < scripts/dev-instance/patron_data.sql```
Would it be alright to have users enter the docker command instead of doing ```docker exec -uroot db bash``` etc?

**Comment by jimchamp** [comment]
Would you be up to creating a `load_patron_data.sh` script, that would run `docker compose exec -T db psql --quiet -U openlibrary openlibrary < scripts/dev-instance/patron_data.sql`, then `docker compose restart memcached` (or a similar restart command)?  Then, developers would need only run the script to load the data.

Otherwise, please do instruct devs to first run `docker compose exec -T db psql --quiet -U openlibrary openlibrary < scripts/dev-instance/patron_data.sql`, then restart memcached.

Thanks!
